### ISSUE TITLE
PartialEq for the structs?

### ISSUE THREAD

**Issue by Ciantic** [aloucks/directx_math#2]
I wonder if it would be feasible to add PartialEq to structs? Not sure how useful it's with floats but it might be.

https://github.com/aloucks/directx_math/blob/master/src/lib.rs#L1594

**Comment by aloucks** [comment]
Those structs are primarily for data load/store. The [XMVector](https://docs.rs/directx_math/0.2.0/directx_math/struct.XMVector.html) wrapper struct does in fact implement `PartialEq`.

https://github.com/aloucks/directx_math/blob/4733743d4b195ca5d7b654e154e0004b6913b258/src/vector.rs#L13097-L13102

However, you probably want to use the [vector2d](https://docs.rs/directx_math/0.2.0/directx_math/vector2d/comparison/index.html), [vector3d](https://docs.rs/directx_math/0.2.0/directx_math/vector3d/comparison/index.html), [vector4d](https://docs.rs/directx_math/0.2.0/directx_math/vector4d/comparison/index.html) comparison functions or use [per-component](https://docs.rs/directx_math/0.2.0/directx_math/vector/comparison/index.html) comparison and check the results with the [utility functions](https://docs.rs/directx_math/0.2.0/directx_math/utility/index.html).

**Comment by Ciantic** [comment]
Thanks! I have been thinking this wrong. I'm new to directx math library (as well as directx 12), and I just noticed that as I started to think about projections, I'm doing a little [Rust DX12 examples](https://github.com/Ciantic/dx12-learning/tree/master/examples), and I find your library fantastic!

If you ever get tired of maintaining it, it would be worth to try get [GFX community](https://github.com/gfx-rs) to adopt it, I think it could be useful bolt in the fast implementation of [wgpu](https://github.com/gfx-rs/wgpu-rs) especially on DX12 backend.
